ui/src/components/TaskDetailsCard/index.jsx

```diff
@@ -91,6 +98,42 @@ export default class TaskDetailsCard extends Component {
   state = {
     showPayload: false,
     showMore: false,
+    Deadlinecopied: false,
+    Expirescopied: false,
+    Createdcopied: false,
+  };
+
+  handleDeadlineCopied = () => {
+    if (this.state.Deadlinecopied) {
+      this.setState({ Deadlinecopied: false });
+    } else {
+      this.setState({ Deadlinecopied: true });
+      setTimeout(() => {
+        this.setState({ Deadlinecopied: false });
+      }, 2000);
+    }
+  };
+
+  handleExpiresCopied = () => {
+    if (this.state.Expirescopied) {
+      this.setState({ Expirescopied: false });
+    } else {
+      this.setState({ Expirescopied: true });
+      setTimeout(() => {
+        this.setState({ Expirescopied: false });
+      }, 2000);
+    }
+  };
+
+  handleCreatedCopied = () => {
+    if (this.state.Createdcopied) {
+      this.setState({ Createdcopied: false });
+    } else {
+      this.setState({ Createdcopied: true });
+      setTimeout(() => {
+        this.setState({ Createdcopied: false });
+      }, 2000);
+    }
   };
```

Comment:
Having to write all of this logic in all of the places we use the copy feature will make us prone for errors. Instead, I think we should create a custom component that handles of this logic. This will make sure we don't re-write the same code multiple times.

Reply:
@helfi92 So I should write the whole function or just the timeout function in another component? And one more thing, in another component how to make it generalize for all the copy buttons as they are independent of others?
_Is there is this.setstate({ ${``}copied :false}) kind of thing in  react ?_

Reply:
No need to do `${``}copied :false`. Since it will be in its own component, we just need to worry of a single state at a time. Perhpas just `this.setState({ copied: false });` for example.

Reply:
So I should write all the functions in another component, not a single one?

Reply:
We want to create a component (e.g., ui/src/components/<new-component>.jsx) then use it in `TaskDetailsCard`, `TaskRunCard`. The component can be used multiple times in the same file.

Reply:
What should be there in the render of the new component?

Reply:
I would have to think about it. What did you try?

Reply:
I haven't tried it yet  , bit busy with semester end exam .what should be in render I was confused about..

Reply:
@helfi92 what are your about it ?

Reply:
@helfi92 What are your views about this?

Reply:
I looked into it. I wrote something. I'll /cc you in the pull-request.

Reply:
https://github.com/taskcluster/taskcluster/pull/2186#issuecomment-566572404